raycaster_float.cpp

```diff
@@ -32,100 +66,90 @@ float RayCasterFloat::Distance(float playerX,
         rayA -= 2.0f * M_PI;
     }
 
-    int tileStepX = 1;
-    int tileStepY = 1;
-    float tileX = 0;
-    float tileY = 0;
-    if (rayA > M_PI) {
-        tileStepX = -1;
-    }
-    if (rayA > M_PI_2 && rayA < 3 * M_PI_2) {
-        tileStepY = -1;
+    float tanA = tan(rayA);
+    float cotA = 1 / tanA;
+    float rayX, rayY, vx, vy;
+    float xOffset, yOffset, vertHitDis, horiHitDis;
+    int depth = 0;
+    const int maxDepth = 100;
```

Comment:
Is it hard-coded?

Reply:
This is the maximum number of attempt for a ray to find a wall. Visually, it can be considered as the depth of field. The value of maxDepth depends on the map size, I think 100 is suitable for current situation.

Reply:
Leave some meaningful comment around this line.

Reply:
Add comment here.

Reply:
ok done